### PR TITLE
Increase Anthropic default max token sampling, prepare for proper support in agents

### PR DESCRIPTION
**Pull request by jacoblee93**
Changes:

- Sets a saner default for Anthropic's max token output which is useful for longer agent responses. 
- Properly support for `stopSequence` passed into `_generate()` in Anthropic used by agents.
- ~~For the zero shot agent, Claude also seemed to think that `[calculator]` was a viable tool rather than `calculator`, so loosened output parsing there~~

### CODE REVIEW

**Comment by jacoblee93** [comment]
TBH Anthropic is *really* slow when used with an agent and a tool like in the test case - probably worth removing for now?

**Comment by nfcampos** [comment]
re: slow what about `claude-instant`?

**Review by nfcampos — COMMENTED** [review]
LGTM, just 2 comments

File: langchain/src/chat_models/anthropic.ts [hunk 1]
```diff
@@ -109,13 +111,13 @@ export class ChatAnthropic extends BaseChatModel implements AnthropicInput {
 
   topP = -1;
 
-  maxTokensToSample = 256;
+  maxTokensToSample = 4096;
```
Comment (nfcampos) [inline langchain/src/chat_models/anthropic.ts:114]: That's a very large number ha, might be worth going for something in between?

File: langchain/src/agents/mrkl/index.ts [hunk 2]
```diff
@@ -112,7 +112,7 @@ export class ZeroShotAgent extends Agent {
     }
 
     return {
-      tool: match[1].trim(),
+      tool: match[1].trim().replace(/\[|\]/g, ""),
```
[the remark this answers is not in the document]
Comment (jacoblee93) [reply langchain/src/agents/mrkl/index.ts:115]: > re: slow what about `claude-instant`?

Genius - unfortunately `instant` isn't smart enough for the zero-shot agent case and just infinitely loops despite getting the correct answer from the tool. I think maybe I'll need to redo the prompt - is it alright if I remove this for now and resubmit another PR later?

[the remark this answers is not in the document]
Comment (nfcampos) [reply langchain/src/agents/mrkl/index.ts]: Yea now worries at all

**Comment by jacoblee93** [comment]
@nfcampos it seems Anthropic with the given prompts isn't really great at parsing tool outputs, so I've removed the agent changes and just kept the default param updates and the `stopSequence` changes. I'll work on redoing some prompts with `PromptSelector` when I have time and submit a bigger PR with it (hopefully) working with all the chat agents.

**Review by nfcampos — APPROVED** [review]
Thanks!

**Pull request merged**